src/shared/wall.c

```diff
@@ -34,97 +36,105 @@ static int write_to_terminal(const char *tty, const char *message) {
         return loop_write_full(fd, message, SIZE_MAX, TIMEOUT_USEC);
 }
 
-#if ENABLE_UTMP
-static int do_wall(
+static int wall_utmp(
         const char *message,
-        const char *username,
-        const char *origin_tty,
         bool (*match_tty)(const char *tty, bool is_local, void *userdata),
         void *userdata) {
 
+#if ENABLE_UTMP
         _unused_ _cleanup_(utxent_cleanup) bool utmpx = false;
         struct utmpx *u;
-        int r;
+        int r = 0;
 
-        utmpx = utxent_start();
+        assert(message);
 
-        r = 0;
+        if (access(_PATH_UTMPX, F_OK) < 0) {
```

Comment:
Maybe add a quick comment here: `/* libc's setutxent() unfortunately doesn't inform us about success, i.e. whether /var/run/utmp exists. Hence we have to check manually first. */`

(simply because doing access() checks like this has this smelly raciness feel)